### ISSUE TITLE
[New game]: Chess game

### ISSUE THREAD

**Issue by rt-001** [kunjgit/GameZone#42]
### 🎮 Game Request

Game logic and basic description
The game is a two-player chess game played on an 8x8 chessboard. Each player has a set of chess pieces and takes turns making moves. Players can select a piece by clicking on it and then choose a valid destination square for that piece. The game enforces the rules of chess, ensuring that moves are legal and capturing opponent's pieces when necessary. The game also handles special moves like castling, en passant, and promotion. It checks for check and checkmate conditions to determine if a player's king is in danger or if the game is over. The game interface is updated to display the current board state, player turn, captured pieces, and other relevant information.

### Point down the features

Game points
1) Flipping the Board: Allow players to flip the chessboard, changing the perspective and orientation of the game. This feature can be useful for players who prefer different viewing angles or want to test their skills from both sides of the board.
2) I will try to implement some cool features such as En Passant, Castling, Threefold Repetition etc.

### Select program in which you are contributing

GSSoC23

### Code of Conduct

- [X] I follow [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md) of this project.

**Comment by kunjgit** [comment]
Hey @rt-001 ! It is really interesting but we want to hear more…..
Can you elaborate the features 🤔!
We are looking forward have your valuable contribution in GameZone 🎮

**Comment by rt-001** [comment]
@kunjgit  Done

**Comment by ManashAnand** [comment]
Hi there! I would like to contribute to this issue. I have gone through the problem description and would love to help out. I have experience with similar problems and believe I can provide a solution. Could you please assign this issue to me so that I can start working on it? Thank you!

**Comment by Lavesh12** [comment]
Hello,
I hope you're doing well. I am interested in contributing to this project as part of GSSoC. It aligns with my skills and interests, and I believe I can make a meaningful contribution to the project Could you please assign this task to me? 
Looking forward to hearing from you.
Thank you,
Lavesh Vishwakarma

**Comment by rt-001** [comment]
@kunjgit assign this to me.

**Comment by rt-001** [comment]
@kunjgit assign this to me.

**Comment by soorjya** [comment]
Hello,
I hope you're doing well. I am interested in contributing to this project as part of GSSoC. It aligns with my skills and interests, and I believe I can make a meaningful contribution to the project Could you please assign this task to me?
Looking forward to hearing from you.
Thank you,

**Comment by kunjgit** [comment]
Hello @rt-001, Time's Uppp!⏰ 
 Sorry for closing your issue! 
 But it's more than a week since we haven't received anything from your side 😢 . 
 Come up with new ideas, create a new issue and make sure you finish it within a week! 🔥 
 All the best! 🚀 
 Happy Hacking! 💗

**Comment by kunjgit** [comment]
Hey @rt-001 ! Thank you so much for your raising the issue💗 
 It’s all yours, you can come anytime again and make some contributions! 🚀 
 Alone, we can do little, but together we can do so much! 😇

**Comment by rt-001** [comment]
@kunjgit Since I am new to GitHub, it was difficult for me to create a new branch for different issues. Usually, I would wait for my current pull request (PR) to get accepted before starting work on another one.
